### PR TITLE
Borcean/client

### PR DESCRIPTION
**Pull request by fahlmant**
Opening a PR to add slave support. I'm sure there is more to do before this can merge, but not sure where to go next.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -16,3 +16,33 @@
...
+
+group 'alfred' do
+  action :create
+end
+
+user 'alfred' do
+  supports manage_home: true
+  gid 'alfred'
+  system true
+  shell '/usr/bin/sh'
```
Comment (borcean) [inline recipes/default.rb]: This should be `/bin/bash` or `/bin/sh` if we choose to specify.
But I think taking the system default would be fine.

Comment (fahlmant) [inline recipes/default.rb]: Don't know why I added /usr in there....

File: .kitchen.cloud.yml [hunk 2]
```diff
@@ -17,14 +17,15 @@ provisioner:
   name: chef_solo
 
 platforms:
-  - name: centos-6.6
-    driver_plugin: openstack
-    driver_config:
-      username: centos
-      image_ref: "CentOS 6.6"
```
Comment (ramereth) [inline .kitchen.cloud.yml]: Any particular reason to remove 6.6 support? We still have quite a bit of 6.6 deployed.

File: recipes/default.rb [hunk 3]
```diff
@@ -16,3 +16,25 @@
...
+  system true
+  shell '/bin/bash'
+  home '/home/alfred'
+  action :create
+end
+
+ohai 'reload_passwd' do
+  action :nothing
+  plugin 'etc'
+end.run_action(:reload)
```
Comment (ramereth) [inline recipes/default.rb]: Can we add a comment here explaining why this is needed (i.e. ssh-keys bug)?

File: .kitchen.cloud.yml (hunk 2 above)
Comment (borcean) [inline .kitchen.cloud.yml]: @ramereth My hindsight, the tests have been readded for CentOS 6.6 :blush:

**Comment by fahlmant** [comment]
@jordane @ramereth Can I get a review please? :)

**Comment by fahlmant** [comment]
@jordane @ramereth ping

File: .kitchen.cloud.yml [hunk 4]
```diff
@@ -22,9 +22,15 @@ platforms:
     driver_config:
       username: centos
       image_ref: "CentOS 6.6"
```
Comment (jordane) [inline .kitchen.cloud.yml]: This should be bumped to 6.7 now.

**Comment by borcean** [comment]
Rebased off master.

File: .kitchen.yml [hunk 5]
```diff
@@ -6,12 +6,43 @@ provisioner:
...
+  - name: centos-6.7
+    driver_plugin: openstack
+    driver_config:
+      username: centos
+      image_ref: "CentOS 6.7"
+  - name: centos-7.1
+    driver_plugin: openstack
+    driver_config:
+      username: centos
+      image_ref: "CentOS 7.1"
```
Comment (ramereth) [inline .kitchen.yml]: You don't need to pull in the openstack configuration in the standard TK file.

File: .kitchen.yml [hunk 6]
```diff
@@ -6,12 +6,43 @@ provisioner:
...
 
 suites:
   - name: default
+    data_bags_path: test/integration/data_bags
```
Comment (ramereth) [inline .kitchen.yml]: You might as well add this to the global config in both TK files.

File: .kitchen.yml [hunk 7]
```diff
@@ -6,12 +6,43 @@ provisioner:
...
     run_list:
       - recipe[osl-jenkins::default]
+  - name: master
+    run_list:
+      - recipe[osl-jenkins::master]
+      - recipe[jenkins::java]
+      - recipe[jenkins::master]
+  - name: haproxy
+    data_bags_path: test/integration/data_bags
+    encrypted_data_bag_secret_key_path: test/integration/encrypted_data_bag_secret
```
Comment (ramereth) [inline .kitchen.yml]: Might as well add this to the global config too.

**Comment by ramereth** [comment]
:+1:

**Pull request merged**